andrada12/assignment3/Dictionary.java

```diff
@@ -0,0 +1,62 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author andrada
+ */
+public class Dictionary {
+    
+    static HashMap<String, ArrayList<String>> dictionary = new HashMap<String, ArrayList<String>>();
+    
+    public static void setDictionary(String[] words){
```

Comment:
Optional: this is a perfectly good data structure, but isn't as space efficient as it could be. Can you do better? This is the sort of follow up question that gets asked at an interview.

If you get stuck you might want to look at https://en.wikipedia.org/wiki/Trie